File: java/src/main/java/com/android/frame/mvp/extra/RecyclerView/CustomLoadMoreView.java
package com.android.frame.mvp.extra.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.android.java.R;
import com.chad.library.adapter.base.loadmore.BaseLoadMoreView;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import org.jetbrains.annotations.NotNull;

/**
 * Created by xuzhb on 2020/12/31
 * Desc:自定义加载更多的脚布局
 */
public class CustomLoadMoreView extends BaseLoadMoreView {

    @NotNull
    @Override
    public View getLoadComplete(@NotNull BaseViewHolder baseViewHolder) {
        return baseViewHolder.getView(R.id.complete_fl);
    }

    @NotNull
    @Override
    public View getLoadEndView(@NotNull BaseViewHolder baseViewHolder) {
        return baseViewHolder.getView(R.id.end_fl);
    }

    @NotNull
    @Override
    public View getLoadFailView(@NotNull BaseViewHolder baseViewHolder) {
        return baseViewHolder.getView(R.id.fail_fl);
    }

    @NotNull
    @Override
    public View getLoadingView(@NotNull BaseViewHolder baseViewHolder) {
        return baseViewHolder.getView(R.id.loading_ll);
    }

    @NotNull
    @Override
    public View getRootView(@NotNull ViewGroup viewGroup) {
        return LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.view_load_more, viewGroup, false);
    }

}

File: kotlin/src/main/java/com/android/frame/mvc/extra/RecyclerView/CustomLoadMoreView.kt
package com.android.frame.mvc.extra.RecyclerView

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.android.basicproject.R
import com.chad.library.adapter.base.loadmore.BaseLoadMoreView
import com.chad.library.adapter.base.viewholder.BaseViewHolder

/**
 * Created by xuzhb on 2020/7/29
 * Desc:自定义加载更多的脚布局
 */
class CustomLoadMoreView : BaseLoadMoreView() {

    override fun getLoadComplete(holder: BaseViewHolder): View = holder.getView(R.id.complete_fl)

    override fun getLoadEndView(holder: BaseViewHolder): View = holder.getView(R.id.end_fl)

    override fun getLoadFailView(holder: BaseViewHolder): View = holder.getView(R.id.fail_fl)

    override fun getLoadingView(holder: BaseViewHolder): View = holder.getView(R.id.loading_ll)

    override fun getRootView(parent: ViewGroup): View =
        LayoutInflater.from(parent.context).inflate(R.layout.view_load_more, parent, false);

}